### PR TITLE
feat: Focus browser from select browser screen and on dashboard login

### PR DESCRIPTION
**Pull request by ryanthemanuel**
- Closes [UNIFY-953](https://cypress-io.atlassian.net/browse/UNIFY-953) and [UNIFY-922](https://cypress-io.atlassian.net/browse/UNIFY-922)

### User facing changelog

n/a

### Additional details

**UNIFY-953**

The goal of this change is to wire up the focus button created in #19830. We will do this by adding a browser action that focuses to the currently opened browser. Note that firefox doesn't allow for setting focus in the browser via automation requests due to focusmanager.testmode set. Therefore we will handle firefox differently. 

For non-firefox, we will hook into the automation that is setup in the server as follows:

* Chrome (or chromium like browsers) - We will use CDP automation to call Page.bringToFront
* Electron - We will tap into the focus automation message and call win.show on the window since we already have access to that where the automation is set up

For firefox, we will use native commands to set focus to the main window in the browser process for MacOS and Windows. On linux, we will do nothing for now. We have decided to scope that combination out (firefox on linux) for the time being.

**UNIFY-922**

For this, we will hook into the dashboard login completion. If the launchpad was in focus previously, we'll set focus back there, otherwise we will call the action created in UNIFY-953.

### PR Tasks

- [X] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Review by ZachJW34 — DISMISSED** [review]
Tested, looks good. #19781 was recently merged that fixed the Electron browser launching which will be good to test against.

**Comment by brian-mann** [comment]
Since we're not supporting focus in linux + firefox, but support every other combination... we probably need to update the graphql for the frontend to "know" whether to render the `Focus Browser` button ahead of time.

**Review by brian-mann — CHANGES_REQUESTED** [review]
Need to update the graphql schema indicating whether the browser is focusable so the frontend knows to render or not render the `Focus Browser` button

File: packages/launchpad/src/setup/OpenBrowserList.vue [hunk 1]
```diff
@@ -154,6 +155,12 @@ mutation OpenBrowserList_SetBrowser($id: ID!) {
 }
 `
 
+gql`
```
Comment (ZachJW34) [inline packages/launchpad/src/setup/OpenBrowserList.vue:158]: Unused, as the `Close` button emits an event to the parent in order to execute the mutation.

File: packages/launchpad/src/setup/OpenBrowserList.vue [hunk 2]
```diff
@@ -204,6 +217,7 @@ const browsers = computed(() => {
 })
 
 const setBrowser = useMutation(OpenBrowserList_SetBrowserDocument)
+const closeBrowserMutation = useMutation(OpenBrowserList_CloseBrowserDocument)
```
Comment (ZachJW34) [inline packages/launchpad/src/setup/OpenBrowserList.vue:220]: Unused.

File: packages/launchpad/src/setup/OpenBrowserList.vue [hunk 3]
```diff
@@ -239,4 +253,10 @@ const browserText = computed(() => {
   }
 })
 
+const focusActiveBrowserWindow = useMutation(OpenBrowser_FocusActiveBrowserWindowDocument)
```
Comment (ZachJW34) [inline packages/launchpad/src/setup/OpenBrowserList.vue:256]: We have two different ways of calling the mutation. Your method calls the mutation in this component, the other emits an event to the parent component to execute the mutation. I'm normally fine with either, but we already emitting for both the launchProject and closeBrowser so let's go with that method. There is also a `TODO` in the `emits` of this component that we can remove for browser focus with this suggested change.

File: packages/server/lib/gui/auth.ts [hunk 4]
```diff
@@ -222,7 +222,7 @@ const start = (onMessage, utmCode) => {
   })
   .finally(() => {
     _internal.stopServer()
-    require('./windows').focusMainWindow()
+    onLogin()
```
Comment (brian-mann) [inline packages/server/lib/gui/auth.ts:225]: yay i'm glad you updated this

**Review by ZachJW34 — APPROVED** [review]
Looks great! LGTM!!

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**